lib/grant.js

```diff
@@ -436,9 +444,18 @@ function saveRefreshToken (done) {
   }
 
   var expires = null;
-  if (this.config.refreshTokenLifetime !== null) {
+
+  var ttl = !isNaN(parseFloat(this.client.refreshTokenLifetime)) && isFinite(this.client.refreshTokenLifetime) ?
+    this.client.refreshTokenLifetime : this.config.refreshTokenLifetime;
+
+  if (ttl !== null) {
+    expires = new Date(this.now);
+    expires.setSeconds(expires.getSeconds() + ttl);
+  }
```

Comment:
I think the same 4 lines are repeated below: https://github.com/mjsalinger/node-oauth2-server/blob/98-Client-Specific-Timeouts/lib/grant.js#L451-L459